blockmanager.go

```diff
@@ -3043,4 +3097,177 @@ func (l *lightHeaderCtx) RelativeAncestorCtx(
+				prevNode *headerlist.Node
+			)
+			// Ensure there is a previous header to compare against.
+			prevNodeEl := b.headerList.Back()
+			if prevNodeEl == nil {
+				log.Warnf("side load - Header list does not contain a previous" +
+					"element as expected -- exiting side load")
+
+				return
+			}
+
+			node = &headerlist.Node{Header: *header}
+			prevNode = prevNodeEl
+			node.Height = prevNode.Height + 1
+
+			if b.cfg.sideLoad.Verify {
+				valid, err := b.verifyBlockHeader(header, *prevNode)
+				if err != nil || !valid {
+					log.Debugf("Side Load- Did not pass verification at height %v"+
+						"-- rolling back to last verified checkpoint and skipping sideload", node.Height)
+
+					prevCheckpoint := b.findPreviousHeaderCheckpoint(
+						node.Height,
+					)
+
+					log.Infof("Rolling back to previous validated "+
+						"checkpoint at height %d/hash %s",
+						prevCheckpoint.Height,
+						prevCheckpoint.Hash)
+
+					//It is possible that the number of headers in the batch writer's buffer is less than the
+					//BlkHeaderWriteBatch and so these headers have not been written into the DB at this time.
+					//We wait for the DBTickerBlkHeaderWriteDuration to expire and write these headers into the
+					//DB before we roll back which should take approximately twice the DBTickerBlkHeaderWriteDuration.
+					//This ensures that these headers are wiped off, during the rollback, in the batch writer's buffer
+					//and in the DB as well.
+					time.Sleep(DBTickerBlkHeaderWriteDuration * 2)
+					err = b.rollBackToHeight(uint32(prevCheckpoint.Height))
+					if err != nil {
+						panic(fmt.Sprintf("Rollback failed: %s", err))
+						// Should we panic here?
+					}
+					tipHeader, height, err := b.cfg.BlockHeaders.ChainTip()
+					if err != nil {
+						return
+					}
+
+					b.headerList.ResetHeaderState(headerlist.Node{
+						Height: int32(height),
+						Header: *tipHeader,
+					})
+					return
+				}
+
+			}
+
+			// Verify checkpoint only if verification is enabled.
+			if b.nextCheckpoint != nil && b.cfg.sideLoad.Verify &&
+				node.Height == b.nextCheckpoint.Height {
+
+				nodeHash := node.Header.BlockHash()
+				if nodeHash.IsEqual(b.nextCheckpoint.Hash) {
+
+					log.Infof("Verified downloaded block "+
+						"header against checkpoint at height "+
+						"%d/hash %s", node.Height, nodeHash)
+				} else {
+					log.Warnf("Error at checkpoint while side loading headers, exiting"+
+						"%d/hash %s", node.Height, nodeHash)
+					return
+				}
+
+			}
+
+			//Add to batch writer.
+			b.blockHeaderBatchWriter.AddItem(headerfs.BlockHeader{
+				BlockHeader: header,
+				Height:      uint32(node.Height),
+			},
+			)
+
+			//Update headerTip to give more accuarate info about tip of DB.
+			b.nextCheckpoint = b.findNextHeaderCheckpoint(node.Height)
+			b.newHeadersMtx.Lock()
+			b.headerTip = uint32(node.Height)
+			b.headerTipHash = node.Header.BlockHash()
+			b.newHeadersMtx.Unlock()
+			b.newHeadersSignal.Broadcast()
```

Comment:
not sure if we should do this after all the headers have been stored in disk instead of after each new header

Reply:
The cfhandler needs it to fetch filter headers, so it needs to be updated. Bock header and filter headers are fetched simultaneously to keep the previous behavior.